include/sdsl/int_vector.hpp

```diff
@@ -286,9 +288,30 @@ class int_vector {
 
 private:
 	size_type	  m_size;  //!< Number of bits needed to store int_vector.
+	size_type	  m_capacity; //!< Number of bits reserved by int_vector.
 	uint64_t*	  m_data;  //!< Pointer to the memory for the bits.
 	int_width_type m_width; //!< Width of the integers.
 
+	// Hidden, since number of bits (size) does not go well together with int value.
+	void bit_resize(const size_type size, const value_type value);
+
+	// this uses a growth factor of 1.5
+	// see explanation in documentation of FBVector
+	// https://github.com/facebook/folly/blob/master/folly/docs/FBVector.md#memory-handling
+	void amortized_resize(const size_type size)
+	{
+		size_type bit_size = size * m_width;
+		if (bit_size > m_capacity || m_data == nullptr) {
+			// start with 64 bit if vector has no capacity
+			size_type tmp_capacity = m_capacity == 0 ? 64 : m_capacity;
+			// find smallest x s.t. m_capacity * 1.5 ** x >= size
+			auto resize_factor = pow(1.5, std::ceil(std::log((bit_size + tmp_capacity - 1) / tmp_capacity) / std::log(1.5)));
```

Comment:
Can we make the 1.5 factor a named constant on the class? Alternatively this could be a template parameter.

Reply:
I agree!

Reply:
@h-2 Any preference which alternative?